fili-core/src/main/java/com/yahoo/bard/webservice/util/SimplifiedIntervalList.java

```diff
@@ -369,4 +370,22 @@ public SimplifiedIntervalList subtract(SimplifiedIntervalList that) {
                 .collect(Collectors.toList());
         return IteratorUtils.chainedIterator(periodIterators);
     }
+
+    /**
+     * Returns an Optional of inner interval with the oldest starting instance.
+     *
+     * @return the Optional of inner interval with the oldest starting instance
+     */
+    public Optional<Interval> getStart() {
+        return stream().min(IntervalStartComparator.INSTANCE);
+    }
+
+    /**
+     * Returns an Optional of inner interval with the latest ending instance.
+     *
+     * @return the Optional of inner interval with the latest ending instance
+     */
+    public Optional<Interval> getEnd() {
+        return stream().max(IntervalEndComparator.INSTANCE);
+    }
```

Comment:
similarly SIL guaranatees that the end of the last interval in the list is the end of the range of time.